pkg/adaptors/authenticateMiddleware.ts

```diff
@@ -0,0 +1,74 @@
+import { Ether, Option } from '@mikuroxina/mini-fn';
+import type { MiddlewareHandler } from 'hono';
+import { createMiddleware } from 'hono/factory';
+
+import {
+  authenticateTokenSymbol,
+  type AuthenticationTokenService,
+} from '../accounts/service/authenticationTokenService.js';
+
+export class AuthenticateMiddlewareService {
+  private readonly authTokenService: AuthenticationTokenService;
+
+  constructor(authTokenService: AuthenticationTokenService) {
+    this.authTokenService = authTokenService;
+  }
+
+  private parseToken(token: string): Option.Option<string> {
+    const split = token.split('.')[1];
+    if (!split) {
+      return Option.none();
+    }
+    const payload = JSON.parse(
+      Buffer.from(split, 'base64').toString('utf-8'),
+    ) as { sub: string };
+    return Option.some(payload.sub);
+  }
+
+  /**
+   *
+   * @param options forceAuthorized: if false, allow request without token
```

Comment:
```suggestion
   * @param {Object} options - Configuration of the middleware.
   * @param {boolean} options.forceAuthorized - Requires an authorization token in the request if true, or not.
```
肯定形のドキュメントにしたほうがよいでしょう. また [JSDoc ではオブジェクトのドキュメントはこのように](https://jsdoc.app/tags-param#parameters-with-properties) 記述します.